### PR TITLE
[Feature][Connector-V2][RocketMQ] Add RocketMQ  source and sink connector

### PR DESCRIPTION
**Pull request by sunxiaojian**
## Purpose of this pull request

[#3018](https://github.com/apache/incubator-seatunnel/issues/3018) RocketMQ source and sink connector

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-translation/seatunnel-translation-flink/src/main/java/org/apache/seatunnel/translation/flink/source/BaseSeaTunnelSourceFunction.java [hunk 1]
```diff
@@ -149,8 +149,10 @@ public void initializeState(FunctionInitializationContext initializeContext) thr
                     getStateName(),
                     Types.MAP(
                         BasicTypeInfo.INT_TYPE_INFO,
-                        Types.LIST(PrimitiveArrayTypeInfo.BYTE_PRIMITIVE_ARRAY_TYPE_INFO))
-                ));
+                        Types.LIST(PrimitiveArrayTypeInfo.BYTE_PRIMITIVE_ARRAY_TYPE_INFO)
```
Comment (hailin0) [inline seatunnel-translation/seatunnel-translation-flink/src/main/java/org/apache/seatunnel/translation/flink/source/BaseSeaTunnelSourceFunction.java]: revert

**Review by hailin0 — COMMENTED** [review]
Add e2e testcase

reference
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e

**Comment by sunxiaojian** [comment]
> Add e2e testcase
> 
> reference https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e

Thanks for reviewing the PR, I will solve it in a week

**Comment by sunxiaojian** [comment]
> > Add e2e testcase
> > reference https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e
 
Added

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/logback-test.xml]: remove logback config file

seatunnel using log4j2

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/rocketmq/rocketmq_source_earliest_to_console.conf [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  }
+}
+
+transform {
+ }
+
+sink  {
+       Console {}
+
+         Assert {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/rocketmq/rocketmq_source_earliest_to_console.conf]: align indent

**Review by hailin0 — COMMENTED** [review]
update [release-note.md](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md)

**Comment by sunxiaojian** [comment]
> update [release-note.md](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md)

fixed,   CI code style check failed,  It can be solved after PR [https://github.com/apache/incubator-seatunnel/pull/4101 ] is merged

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/logback-test.xml (hunk 2 above)
Comment (sunxiaojian) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/logback-test.xml]: > remove logback config file
> 
> seatunnel using log4j2

To disable the debug log
https://www.testcontainers.org/supported_docker_environment/logging_config/

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/rocketmq/rocketmq_source_earliest_to_console.conf (hunk 3 above)
Comment (sunxiaojian) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/rocketmq/rocketmq_source_earliest_to_console.conf]: fixed

Comment (sunxiaojian) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/rocketmq/rocketmq_source_earliest_to_console.conf]: > align indent

fixed

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/logback-test.xml (hunk 2 above)
Comment (sunxiaojian) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/logback-test.xml]: > > remove logback config file
> > seatunnel using log4j2
> 
> To disable the debug log https://www.testcontainers.org/supported_docker_environment/logging_config/

@hailin0  At present, all tests have been completed. Logback-test is mainly to disable some debug logs. Do you want to continue to modify here? Or there is a better way to deal with it

Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/logback-test.xml]: logback is adapted to log4j2 through slf4j bridge, you can configure `log4j2-test.properties` logger
@sunxiaojian

Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/logback-test.xml]: The easiest way is to delete the logback configuration

Comment (sunxiaojian) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rocketmq-e2e/src/test/resources/logback-test.xml]: > logback is adapted to log4j2 through slf4j bridge, you can configure `log4j2-test.properties` logger @sunxiaojian

fixed

**Comment by sunxiaojian** [comment]
> ## Purpose of this pull request
> [#3018](https://github.com/apache/incubator-seatunnel/issues/3018) RocketMQ source and sink connector
> 
> ## Check list
> * [ ]  Code changed are covered with tests, or it does not need tests for reason:
> * [ ]  If any new Jar binary package adding in your PR, please add License Notice according
>   [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
> * [ ]  If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
> * [ ]  If you are contributing the connector code, please check that the following files are updated:
>   
>   1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
>   2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
>   3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
> * [ ]  Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

@hailin0 PTAL

**Comment by sunxiaojian** [comment]
@hailin0 PTAL

**Comment by sunxiaojian** [comment]
@hailin0 @EricJoy2048 PTAL

**Comment by sunxiaojian** [comment]
@TyrantLucifer

**Review by TyrantLucifer — COMMENTED** [review]
Overall LGTM, You should update error codes in this document: https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/connector-v2/Error-Quick-Reference-Manual.md

**Comment by sunxiaojian** [comment]
> Overall LGTM, You should update error codes in this document: https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/connector-v2/Error-Quick-Reference-Manual.md

@TyrantLucifer  done

**Review by TyrantLucifer — APPROVED** [review]
LGTM, thank you for your contribution!

**Review by hailin0 — APPROVED** [review]

**Pull request merged**